### PR TITLE
backup: init restore controller

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind cleanup

/kind feature

**What this PR does / why we need it**:

init the restore controller.
part of #404

add ut for restore controller helper function：

kurator.dev/kurator/pkg/fleet-manager/backup_restore_migrate_shared.go:325:	fetchRestoreDestinationClusters		0.0%
kurator.dev/kurator/pkg/fleet-manager/backup_restore_migrate_shared.go:368:	isFleetClusterSubset			100.0%
kurator.dev/kurator/pkg/fleet-manager/backup_restore_migrate_shared.go:378:	buildVeleroRestoreInstance		93.3%
kurator.dev/kurator/pkg/fleet-manager/backup_restore_migrate_shared.go:410:	allRestoreCompleted			100.0%
kurator.dev/kurator/pkg/fleet-manager/backup_restore_migrate_shared.go:421:	syncVeleroRestoreStatus			0.0%

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
init restore controller
```

### CODE REVIEW

**Comment by Xieql** [comment]
@hzxuzhonghu PTAL

File: pkg/fleet-manager/restore_controller.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	return r.reconcileRestore(ctx, restore)
+}
+
+// reconcileRestore handles the main reconcile logic for a Restore object.
+func (r *RestoreManager) reconcileRestore(ctx context.Context, restore *backupapi.Restore) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	fleetName, destinationClusters, err := r.fetchRestoreDestinationClusters(ctx, restore)
+	if err != nil {
+		log.Error(err, "failed to fetch destination clusters for restore", "restoreName", restore.Name)
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/restore_controller.go:91]: nit: donot you have key values in L50 	`log := ctrl.LoggerFrom(ctx).WithValues("restore", req.NamespacedName)`

**Review by hzxuzhonghu — COMMENTED** [review]
Overall LGTM, except the log has redundant key values

File: pkg/fleet-manager/restore_controller.go (hunk 1 above)
Comment (Xieql) [reply pkg/fleet-manager/restore_controller.go:91]: yes， forgot it

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**